aws/data_source_aws_kinesis_stream_consumer.go

```diff
@@ -0,0 +1,57 @@
+package aws
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceAwsKinesisStreamConsumer() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsKinesisStreamConsumerRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"stream_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"creation_timestamp": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsKinesisStreamConsumerRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).kinesisconn
+	cn := d.Get("name").(string)
+	sa := d.Get("stream_arn").(string)
+
+	state, err := readKinesisStreamConsumerState(conn, cn, sa)
```

Comment:
we can expand the search functionality (by name and/or arn, as well as neither) by using the `ListStreamConsumers` API method, so I'll update here w/usage